pkg/network/ovn_kubernetes.go

```diff
@@ -666,22 +696,55 @@ func bootstrapOVNConfig(conf *operv1.Network, kubeClient cnoclient.Client, hc *H
 	}
 
 	cm := &corev1.ConfigMap{}
-	dmc := types.NamespacedName{Namespace: "openshift-network-operator", Name: "dpu-mode-config"}
+	dmc := types.NamespacedName{Namespace: "openshift-network-operator", Name: "smart-nic-dpu-mode-config"}
 	err = kubeClient.ClientFor("").CRClient().Get(context.TODO(), dmc, cm)
 
 	if err != nil {
 		if !apierrors.IsNotFound(err) {
 			return nil, fmt.Errorf("Could not determine Node Mode: %w", err)
 		}
 	} else {
-		nodeModeOverride := cm.Data["mode"]
-		if nodeModeOverride != OVN_NODE_MODE_DPU_HOST && nodeModeOverride != OVN_NODE_MODE_DPU {
-			klog.Warningf("dpu-mode-config does not match %q or %q, is: %q. Using OVN configuration: %+v",
-				OVN_NODE_MODE_DPU_HOST, OVN_NODE_MODE_DPU, nodeModeOverride, ovnConfigResult)
+		dpuHostModeLabel, exists := cm.Data["dpu-host-mode-label"]
+		if exists {
+			ovnConfigResult.DpuHostModeLabel = dpuHostModeLabel
+		}
+
+		dpuModeLabel, exists := cm.Data["dpu-mode-label"]
+		if exists {
+			ovnConfigResult.DpuModeLabel = dpuModeLabel
+		}
+
+		smartNicModeLabel, exists := cm.Data["smart-nic-mode-label"]
+		if exists {
+			ovnConfigResult.SmartNicModeLabel = smartNicModeLabel
+		}
+
+		resourceName, exists := cm.Data["resource-name"]
+		if exists {
+			ovnConfigResult.ResourceName = resourceName
 		} else {
-			ovnConfigResult.NodeMode = nodeModeOverride
-			klog.Infof("Overriding OVN configuration to %+v", ovnConfigResult)
+			// Resource name must be provided since ovn-kube-node will require resources such as VFs/SFs
```

Comment:
Having seconds thoughts about making this mandatory. Would like to see what everyone thinks.